### PR TITLE
test: cover report and insights read paths

### LINKED ISSUE #147: [b2-mcp] Cover report-client and insights read paths (weak branch coverage)
## Objective

Cover the usage-report and insights read paths, which have weak branch coverage: `src/b2/report-client.ts` (71.4% lines, 52.5% branches, 42 of 134 uncovered) and `src/b2/insights.ts` (72.2% branches, 74 of 509 uncovered).

## Scope of work

Add deterministic unit tests (fake S3/report client, no live B2) for:

- `report-client.ts`: reading daily usage-report CSVs, malformed or empty CSV rows, a missing report bucket, pagination and continuation, the SSRF and host guard rejection branches, and error mapping to the MCP error shape.
- `insights.ts`: the branch paths in `b2_usage_growth`, `b2_egress_leaders`, `b2_largest_files`, and `b2_unfinished_uploads` (empty result sets, time-window filters, ranking and limit boundaries, and bucket-resolution failures).

## Acceptance criteria

- `src/b2/report-client.ts` reaches at least 85% lines and 75% branches; insights branch coverage improves materially.
- Deterministic and credential-free.
- Full `pnpm run verify` green.

References: coverage report on `main`, `tests/unit/report-s3-ssrf.unit.test.ts`.

### PR DESCRIPTION
**Pull request by goanpeca**
## Summary
- Add deterministic B2ReportClient tests for report listing options, body readers, truncation, aborted reads, and malformed or missing bodies.
- Add fake report/native-client insight read path tests covering missing report buckets, pagination, malformed or empty CSVs, ranking and limit boundaries, time-window filters, bucket-resolution failures, timeout branches, and MCP error mapping.
- Stabilize the new insight tests with a fixed system clock, typed report/native fakes, a looser empty-snapshot scan metadata assertion, and an explicit `startPartNumber` pagination assertion.
- Align review coverage tests with the exported contracts by mocking the `createReportS3Client`/`B2S3PeerClient` boundary, removing an unused `getConfig` helper return, and documenting the scan-budget overrun fixture.

## Linked Issue
Closes #147

## Tests
- pnpm run test:unit
- pnpm run verify
- pnpm run test:coverage
- pnpm exec vitest run --config vitest.config.mts --project=unit tests/unit/insights-read-paths.unit.test.ts --coverage=false
- pnpm exec vitest run --config vitest.config.mts --project=unit tests/unit/report-client.unit.test.ts tests/unit/insights-read-paths.unit.test.ts --coverage=false
- pnpm run typecheck
- pnpm run format:check

## Follow-up Notes
- Coverage target result: src/b2/report-client.ts 96.63% lines / 84.74% branches; src/b2/insights.ts 94.59% lines / 83.85% branches.

### CODE REVIEW

File: tests/unit/insights-read-paths.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,489 @@
...
+
+    const result = parseResult(
+      await tools.call("b2_unfinished_uploads", { bucket: "photos", max_uploads: 10 }),
+    );
+
+    expect(result.truncated).toBe(false);
+    expect(result.unfinished_count).toBe(1);
+    expect(result.wasted_gb).toBe(1.5);
+    expect(result.oldest_file).toBe("old.bin");
+    expect(nativeClient.listParts).toHaveBeenCalledTimes(2);
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply tests/unit/insights-read-paths.unit.test.ts:554]: Fixed: Added an assertion that the second listParts request forwards startPartNumber: 2.

File: tests/unit/insights-read-paths.unit.test.ts [hunk 2]
```diff
@@ -0,0 +1,489 @@
...
+  };
+}
+
+function part(contentLength: number, partNumber = 1) {
+  return { partNumber, contentLength };
+}
+
+afterEach(() => {
+  vi.restoreAllMocks();
+});
```
[the remark this answers is not in the document]
Comment (goanpeca) [reply tests/unit/insights-read-paths.unit.test.ts:192]: Fixed: Froze the usage-report tests with fake system time so fixture and handler dates share one clock.
